pkg/chartutil/expand.go

```diff
@@ -81,3 +81,67 @@ func ExpandFile(dest, src string) error {
 	defer h.Close()
 	return Expand(dest, h)
 }
+
+// Expand uncompresses and extracts a chart into the specified directory.
+// Return the directory when extracting successfully.
+func ExpandWithPath(dir string, r io.Reader) (string, error) {
+	gr, err := gzip.NewReader(r)
```

Comment:
I'm not sure I understand so I just want to ask for a bit more clarification. Sorry!

Do you mean that you will refactor the code to have overall less code duplication?